flink-tests/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterStopWithSavepointITCase.java

```diff
@@ -152,7 +152,8 @@ private void throwingExceptionOnCallbackWithoutRestartsHelper(final boolean term
 
         // verifying that we actually received a synchronous checkpoint
         assertTrue(syncSavepointId.get() > 0);
-        assertThat(getJobStatus(), equalTo(JobStatus.FAILED));
+        assertThat(
```

Comment:
Why did that work before? It's not clear to me why this is needed now.

Reply:
It depends how fast the job is being brought down. Apparently on Azure it takes longer for the job to actually `FAIL` and we might query the status when it is still in a `FAILLING` state.

Reply:
There is a similar pattern in `throwingExceptionOnCallbackWithRestartsHelper` where we wait for either `CANCELED` or `CANCELLING`.

Reply:
Okay makes sense.